File: C#.Net/Tests/JDI Tests/TestBase/TestsBase.cs
using Epam.JDI.Core.Settings;
using JDI_Commons;
using JDI_Tests.Epam_UIObjects;
using JDI_Web.Selenium.DriverFactory;
using JDI_Web.Selenium.Elements.Composite;
using JDI_Web.Settings;
using NUnit.Framework;

namespace JDI_Tests.Tests.Epam
{
    [SetUpFixture]
    public class TestsBase
    {
        Timer _timer = new Timer();
        
        [OneTimeSetUp]
        public void Init()
        {
            WebSettings.InitNUnitDefault();
            JDISettings.Logger.Info("Init test run");
            WinProcUtils.KillAllRunWebDrivers();
            _timer = new Timer();
            WebSite.Init(typeof(EpamSite));
            EpamSite.HomePage.Open();
            JDISettings.Logger.Info("Run Tests");
        }

        [OneTimeTearDown]
        public void TestCleanup()
        {
            JDISettings.Logger.Info($@"
Test run finished.
Total test run time: {_timer.TimePassed.ToString(@"hh\:mm\:ss\.fff")}");
            WinProcUtils.KillAllRunWebDrivers();
        }
    }
    
}

File: Java/Tests/jdi-uitest-tutorialtests/src/test/java/com/epam/jdi/uitests/testing/TestsBase.java
package com.epam.jdi.uitests.testing;

import com.epam.jdi.site.epam.EpamSite;
import com.epam.jdi.uitests.web.selenium.driver.WebDriverUtils;
import com.epam.jdi.uitests.web.settings.WebSettings;
import com.epam.jdi.uitests.web.testng.testRunner.TestNGBase;
import org.testng.annotations.AfterSuite;
import org.testng.annotations.BeforeSuite;

import java.io.IOException;

import static com.epam.jdi.uitests.core.logger.LogLevels.STEP;
import static com.epam.jdi.uitests.core.settings.JDISettings.logger;
import static com.epam.jdi.uitests.web.selenium.elements.composite.WebSite.init;

/** Created by Roman_Iovlev on 7/13/2015. */
public abstract class TestsBase extends TestNGBase {

  @BeforeSuite(alwaysRun = true)
  public static void setUp() throws IOException {
    logger.setLogLevel(STEP);
    init(EpamSite.class);
    EpamSite.open();
    logger.info("Run Tests");
  }

  @AfterSuite(alwaysRun = true)
  public static void tearDown() throws IOException {

    if (WebSettings.killBrowser.toLowerCase().contains("after")) {
      WebDriverUtils.killWebBrowserByName(WebSettings.getDriverFactory().currentDriverName());
      // killAllRunWebBrowsers();
    }
  }
}
